pkg/ccl/testccl/sqlccl/sqlccl_test: TestShowTransferState failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestShowTransferState [failed](https://mesolite.cluster.engflow.com/invocations/default/ce0b4fde-7eb8-479d-bd0e-5dbc1a67362e?testReportRun=1&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [5e1ce54bba69fdf695c449920a86fbe6d3613440](https://github.com/cockroachdb/cockroach/commits/5e1ce54bba69fdf695c449920a86fbe6d3613440):


```
=== RUN   TestShowTransferState
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestShowTransferState1836970504
    test_log_scope.go:76: use -show-logs to present logs inline
    show_transfer_state_test.go:279: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTransferState1836970504
--- FAIL: TestShowTransferState (3.44s)
=== RUN   TestShowTransferState/successful_transfer
    show_transfer_state_test.go:162: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:162
        	Error:      	Received unexpected error:
        	            	failed to connect to ``host=127.0.0.1 user=testuser database=``: server error (ERROR: session revival tokens are not supported on this cluster (SQLSTATE 28000))
        	Test:       	TestShowTransferState/successful_transfer
    --- FAIL: TestShowTransferState/successful_transfer (0.00s)
=== RUN   TestShowTransferState/with_transfer_key
    show_transfer_state_test.go:141: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:141
        	Error:      	Should be false
        	Test:       	TestShowTransferState/with_transfer_key
    --- FAIL: TestShowTransferState/with_transfer_key (0.01s)
=== RUN   TestShowTransferState/without_transfer_key
    show_transfer_state_test.go:77: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:77
        	Error:      	Should be false
        	Test:       	TestShowTransferState/without_transfer_key
        	Messages:   	expected null error, got session revival tokens are not supported on this cluster
    --- FAIL: TestShowTransferState/without_transfer_key (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTransferState.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43586

pkg/ccl/testccl/sqlccl/sqlccl_test.TestShowTransferState [failed](https://tanzanite.cluster.engflow.com/invocations/default/3ac05183-1d45-4001-87be-97bd2abe9679?testReportRun=12&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [1b5c41939197efb0fa50ded9795e2e83f5c1dd34](https://github.com/cockroachdb/cockroach/commits/1b5c41939197efb0fa50ded9795e2e83f5c1dd34):


```
=== RUN   TestShowTransferState
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestShowTransferState945568174
    test_log_scope.go:76: use -show-logs to present logs inline
    show_transfer_state_test.go:279: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTransferState945568174
--- FAIL: TestShowTransferState (2.96s)
=== RUN   TestShowTransferState/successful_transfer
    show_transfer_state_test.go:162: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:162
        	Error:      	Received unexpected error:
        	            	failed to connect to ``host=127.0.0.1 user=testuser database=``: server error (ERROR: session revival tokens are not supported on this cluster (SQLSTATE 28000))
        	Test:       	TestShowTransferState/successful_transfer
    --- FAIL: TestShowTransferState/successful_transfer (0.01s)
=== RUN   TestShowTransferState/with_transfer_key
    show_transfer_state_test.go:141: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:141
        	Error:      	Should be false
        	Test:       	TestShowTransferState/with_transfer_key
    --- FAIL: TestShowTransferState/with_transfer_key (0.02s)
=== RUN   TestShowTransferState/without_transfer_key
    show_transfer_state_test.go:77: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:77
        	Error:      	Should be false
        	Test:       	TestShowTransferState/without_transfer_key
        	Messages:   	expected null error, got session revival tokens are not supported on this cluster
    --- FAIL: TestShowTransferState/without_transfer_key (0.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTransferState.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go
+++ b/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go
@@ -51,6 +51,20 @@ func TestShowTransferState(t *testing.T) {
 	_, err = tenantDB.Exec("GRANT SELECT ON tab TO testuser")
 	require.NoError(t, err)
 
+	testutils.SucceedsSoon(t, func() error {
+		// Waiting for the cluster setting to propagate to the tenant.
+		var enabled bool
+		if err := tenantDB.QueryRow(
+			`SHOW CLUSTER SETTING server.user_login.session_revival_token.enabled`,
+		).Scan(&enabled); err != nil {
+			return err
+		}
+		if !enabled {
+			return errors.New("cluster setting not yet propagated")
+		}
+		return nil
+	})
+
 	testUserConn := tenant.SQLConn(t, serverutils.User(username.TestUser))
 
 	t.Run("without_transfer_key", func(t *testing.T) {
@@ -211,19 +225,7 @@ func TestShowTransferState(t *testing.T) {
 		t.Run("root_user", func(t *testing.T) {
 			var key string
 			var errVal, sessionState, sessionRevivalToken gosql.NullString
-			testutils.SucceedsSoon(t, func() error {
-				// Waiting for the cluster setting to propagate to the tenant.
-				var enabled bool
-				if err := tenantDB.QueryRow(
-					`SHOW CLUSTER SETTING server.user_login.session_revival_token.enabled`,
-				).Scan(&enabled); err != nil {
-					return err
-				}
-				if !enabled {
-					return errors.New("cluster setting not yet propagated")
-				}
-				return nil
-			})
+
 			err := tenantDB.QueryRow(`SHOW TRANSFER STATE WITH 'bar'`).Scan(&errVal, &sessionState, &sessionRevivalToken, &key)
 			require.NoError(t, err)
 			require.True(t, errVal.Valid)

